packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -196,9 +240,13 @@ export const plugin = <TContext>(
 
         executionDidStart(requestContext) {
           return {
-            executionDidEnd: () => didEnd(requestContext),
+            executionDidEnd: () => didEnd(requestContext, true),
             willResolveField({ info }) {
-              return treeBuilder.willResolveField(info);
+              // We can early abort if traceReporting returned false
```

Comment:
Or we could just not include the willResolveField method on this object in that case, because this should be either always or never called for a given request?

Reply:
I'm not entirely sure how to not include the `willResolveMethod` after we return the object from `requestDidStart`. I think I might be missing something.

Reply:
```
          const executionListener = {
            executionDidEnd: () => didEnd(requestContext, true),
          };
          if (!endDone) {
            executionListener.willResolveField = ({info}) => ...
          }
          return executionListener;
```

There might be a microscopic performance improvement here, and it makes it clear to readers that the value of endDone is constant across all fields.